notebooks-src/notebooks/R Examples/Comparative Metagenomics.ipynb

```diff
@@ -0,0 +1,1339 @@
+   "id": "4291a3bb-bc92-4762-8ec7-696ff9ee8597",
+   "metadata": {},
+   "source": [
+    "1. Data normalisation. `SIAMCAT` offers a few normalization approaches that can be useful for subsequent statistical modeling in the sense that they transform features in a way that can increase the accuracy of the resulting models. Importantly, these normalization techniques do not make use of any label information, and can thus be applied up front to the whole data set and outside of the following cross validation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a68ff86c-9439-491a-83c5-3dc87fc99fa0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "siamcat_obj = normalize.features(siamcat_obj, norm.method='log.std', norm.param=list(log.n0=1e-06, sd.min.q=0))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a1975ac8-82d7-43d0-b4f5-7a2e7a4c48cf",
+   "metadata": {},
+   "source": [
+    "2. Prepare cross-validation. Cross validation is a technique to assess how well a ML model would generalize to external data by partionining the dataset into training and test sets. `SIAMCAT` greatly simplifies the set-up of cross-validation schemes, including stratification of samples or keeping samples inseperable based on metadata. Here, we split the dataset into 10 parts and then train a model on 9 of these parts and use the left-out part to test the model. The whole process is repeated 10 times."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fffc110e-7497-4d6e-91d9-50a6e9df8690",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "siamcat_obj = create.data.split(siamcat_obj, num.folds=10, num.resample=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2dae8028-2d8d-422b-9c7c-ccf5dcaf2bd8",
+   "metadata": {},
+   "source": [
+    "3. Model training. The actual model training is performed using the function `train.model`. Again, multiple options for customization are available, ranging from the machine learning method to the measure for model selection or customizable parameter set for hyperparameter tuning. Here, we train a Lasso model ([10](#reference_10))."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1914806d-024e-4601-88f4-c4cfbc975d51",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "siamcat_obj = train.model(siamcat_obj, method='lasso')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a251abd4-5fa0-49d0-b637-865646215480",
+   "metadata": {},
+   "source": [
+    "4. Make predictions. This function will automatically apply the models trained in cross validation to their respective test sets and aggregate the predictions across the whole data set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ecf48ffc-5ffa-4e56-a80e-ee07d56ab067",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "siamcat_obj = make.predictions(siamcat_obj)"
```

Comment:
Could add `verbose = F` argument to this one, since it prints a lot of progress bar lines but actually runs very fast